bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -945,50 +945,124 @@ string_impls_for!(Lockdown, "Lockdown");
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// ApiclientCommand represents a valid Bootstrap Command. It stores the command as a vector of
+/// strings and ensures that the first argument is apiclient.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Default, Serialize)]
+pub struct ApiclientCommand(Vec<String>);
+
+impl ApiclientCommand {
+    pub fn get_command_and_args(&self) -> (&str, &[String]) {
+        self.0
+            .split_first()
+            .map(|(command, rest)| (command.as_str(), rest))
+            .unwrap_or_default()
+    }
+}
+
+// Custom deserializer added to enforce rules to make sure the command is valid.
+impl<'de> serde::Deserialize<'de> for ApiclientCommand {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let original: Vec<String> = serde::Deserialize::deserialize(deserializer)?;
+        Self::try_from(original).map_err(|e| {
+            <D::Error as serde::de::Error>::custom(format!(
+                "Unable to deserialize into ApiclientCommand: {}",
+                e
+            ))
+        })
+    }
+}
+
+impl TryFrom<Vec<String>> for ApiclientCommand {
+    type Error = error::Error;
+
+    fn try_from(input: Vec<String>) -> std::result::Result<Self, error::Error> {
+        let first_word = input.first().map(String::as_str);
+        ensure!(
+            matches!(first_word, Some("apiclient")),
+            error::InvalidCommandSnafu { input },
+        );
+
+        Ok(ApiclientCommand(input))
+    }
+}
+
+#[cfg(test)]
+mod test_valid_apiclient_command {
+    use super::ApiclientCommand;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_apiclient_command() {
+        assert!(ApiclientCommand::try_from(vec![
+            "apiclient".to_string(),
+            "motd=helloworld".to_string()
```

Comment:
nit: this isn't a valid command if you actually tried it though right? `apiclient motd=helloworld` will result in an error, it probably needs to be
```suggestion
            "get".to_string(),
            "motd=helloworld".to_string()
``` to actually be a valid command?

Reply:
Updated. Thank you for spotting that.